libs/ledger-live-common/src/families/tron/api/index.ts

```diff
@@ -140,10 +141,26 @@ export const createTronTransaction = async (
       ? `${getBaseApiUrl()}/wallet/transferasset`
       : `${getBaseApiUrl()}/wallet/createtransaction`;
     const preparedTransaction = await post(url, txData);
-    return preparedTransaction;
+    // for the ledger Vault we need to increase the expiration
+    return extendTronTxExpirationTimeBy10mn(preparedTransaction);
   }
 };
 
+function extendTronTxExpirationTimeBy10mn(
+  preparedTransaction,
+): Promise<SendTransactionDataSuccess> {
+  const VAULT_EXPIRATION_TIME = 600;
+  const HttpProvider = TronWeb.providers.HttpProvider;
+  const fullNode = new HttpProvider(getBaseApiUrl());
+  const solidityNode = new HttpProvider(getBaseApiUrl());
+  const eventServer = new HttpProvider(getBaseApiUrl());
+  const tronWeb = new TronWeb(fullNode, solidityNode, eventServer);
+  // tronWeb.setHeader({
+  //   "TRON-PRO-API-KEY": "ec320f10-8bd8-4ac9-a6cf-eef9bd346e25",
+  // });
+  return tronWeb.transactionBuilder.extendExpiration(preparedTransaction, VAULT_EXPIRATION_TIME);
```

Comment:
is this essentially doing a http call?
since the tron implementation generally didn't use "tronweb" i think it would be good to try to do the same without having to depends on tronweb (if it's possible to do it with api calls) so it stays consistent with the coin implementation.

it's on @LedgerHQ/live-blockchain-support  to review / advise further now 🙏

Reply:
Very good point but we had a look and at the moment there doesn't seem to be a way to do that with the HTTP API, so TronWeb is the most straightforward to do it...

Reply:
Actually after a deeper look in the tronweb lib, it looks like the only part that's useful to you is [this](https://github.com/tronprotocol/tronweb/blob/4b131bbcaa94f9d70fffd1dc36577703d42eefa1/src/lib/transactionBuilder.js#L2690C1-L2693C66).

@drakoFukayu could you please try using these few lines instead of the whole tronweb lib?

Sorry for asking for more work, the rationale is that the bundle size for Live is already very large, causing long build and loading time, which will be even more of an issue when we want to move to a web environment.
So we are trying to add new dependencies only when they're absolutely necessary.

Reply:
Needs more than the few lines I mentioned, doesn't seem there is an alternative to using tronweb at the moment.

For context: I have contacted trongrid who are willing to add support for changing the expiration through an HTTP call, so we may consider an evolution soon to remove the tronweb lib and just use the HTTP API.